### ISSUE TITLE
Crash on initialization if showLogoBug is NO

### ISSUE THREAD

**Issue by jklimke** [mapbox/DEPRECATED-mapbox-ios-sdk#393]
Hi,

i just upgraded my mapbox ios version to use the latest release version. here it happens that the RMMapView can not be initialized correctly.

The application crashes with the follwoing error (left something out):

```
The view hierarchy is not prepared for the constraint: <NSLayoutConstraint:0xa2ca290 V:[UIImageView:0xa277010]-(-564)-[_UILayoutGuide:0xa2c9af0]>
    When added to a view, the constraint's items must be descendants of that view (or the view itself). This will crash if the constraint needs to be resolved before the view hierarchy is assembled. Break on -[UIView _viewHierarchyUnpreparedForConstraint:] to debug.
2014-02-18 15:30:04.106 SmartMap[31943:70b] View hierarchy unprepared for constraint.
```

This happens on updating the layout constraints in RMMapView updateConstraints method. Since i configured the showLogoBug property to be set to NO, the _logoBug UIView is not part of the hierarchy, which causes the crash.

```
 if (_logoBug)
    {
        if ( ! [[viewController.view valueForKeyPath:@"constraints.firstItem"]  containsObject:_logoBug] &&
             ! [[viewController.view valueForKeyPath:@"constraints.secondItem"] containsObject:_logoBug])
        {
            CGFloat leftSpacing   = _logoBug.frame.origin.x;
            CGFloat bottomSpacing = _logoBug.superview.bounds.size.height - _logoBug.frame.origin.y - _logoBug.bounds.size.height;

            NSString *formatString;
            NSDictionary *views;

            if (RMPostVersion7)
            {
                formatString = @"V:[logoBug]-bottomSpacing-[bottomLayoutGuide]";
                views = @{ @"logoBug" : _logoBug,
                           @"bottomLayoutGuide" : viewController.bottomLayoutGuide };
            }
            else
            {
                formatString = @"V:[logoBug]-bottomSpacing-|";
                views = @{ @"logoBug" : _logoBug };
            }

            [viewController.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:formatString
                                                                                        options:0
                                                                                        metrics:@{ @"bottomSpacing" : @(bottomSpacing) }
                                                                                          views:views]];

            [viewController.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-leftSpacing-[logoBug]"
                                                                                        options:0
                                                                                        metrics:@{ @"leftSpacing" : @(leftSpacing) }
                                                                                          views:views]];
        }
    }
```

**Comment by incanus** [comment]
@jklimke can you check if this is solved for you as of 1cd588119524925f9c2eb1094da06f65223766e9?

**Comment by jklimke** [comment]
Yes ... This one works for me. Thanks for your quick reply.

**Comment by incanus** [comment]
Cool, that's in `develop` now and will make the next release. The `develop` branch is quite stable at the moment as mostly bug fixes over `1.1.0` have been going into it of late.
